### ISSUE TITLE
If seed is fixed, SkoptSampler suggests exactly the same parameter sets during startup trials

### ISSUE THREAD

**Issue by rene-rex** [optuna/optuna#4022]
### Expected behavior

The suggested parameters should be reproducible but not all trials should use exactly the same values.

### Environment

- Optuna version:3.0.2
- Python version:3.8.5
- OS:Windows-10-10.0.19041-SP0
- skopt version: 0.9.0

### Error messages, stack traces, or logs

```shell
params_x
0    0.976270
1    1.856892
2    1.856892
3    1.856892
4    1.856892
5    1.856892
6    1.856892
7    1.856892
8    1.856892
9    1.856892
10   2.326864
11  -9.084558
12  10.000000
13   2.173857
14   2.111044
15   2.090603
16   2.069437
17   2.061775
18   2.032221
19   2.018462
```

### Steps to reproduce

Run the minimal example. Trials 1 (not 0) to 9 have identical x.

```python
import optuna


def objective(trial):
    x = trial.suggest_float("x", -10, 10)
    return (x - 2) ** 2

sampler = optuna.integration.SkoptSampler(seed=0)
study = optuna.create_study(sampler=sampler)

study.optimize(objective,  n_trials=20)

print(study.trials_dataframe()[['params_x']])
```

![image](https://user-images.githubusercontent.com/7721866/192567723-5362ec5e-26d7-47cc-a236-93971c54ae23.png)

**Comment by not522** [comment]
Thank you for your bug report. I investigated the cause and it will be fixed by #4023.

**Comment by rene-rex** [comment]
Thanks, that was really quick! I tried your branch from the merge request and it solves my issue indeed.
